libs/MeshKernel/include/MeshKernel/AveragingInterpolation.hpp

```diff
@@ -95,7 +95,7 @@ namespace meshkernel
         /// @param[in] useClosestSampleIfNoneAvailable If no samples are found, use the closest one.
         /// @param[in] subtractSampleValues            For some algorithms (e.g. refinement based on levels) we need to subtract 1 to the sample value.
         /// @param[in] minNumSamples                   The minimum a of samples used for certain interpolation algorithms
-        AveragingInterpolation(std::shared_ptr<Mesh2D> mesh,
+        AveragingInterpolation(Mesh2D& mesh,
```

Comment:
AveragingInterpolation does not seem to modify m_mesh. Could you pass mesh by const ref to the constructor.

Reply:
unfortunately the reference is modified in one of the methods